test/com/amazon/ion/impl/IonCursorBinaryTest.java

```diff
@@ -0,0 +1,252 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.ion.impl;
+
+import com.amazon.ion.IonBufferConfiguration;
+import com.amazon.ion.IonCursor;
+import com.amazon.ion.IvmNotificationConsumer;
+import org.junit.Before;
+import org.junit.Test;
+
+import static com.amazon.ion.BitUtils.bytes;
+import static com.amazon.ion.IonCursor.Event.END_CONTAINER;
+import static com.amazon.ion.IonCursor.Event.NEEDS_DATA;
+import static com.amazon.ion.IonCursor.Event.NEEDS_INSTRUCTION;
+import static com.amazon.ion.IonCursor.Event.VALUE_READY;
+import static com.amazon.ion.IonCursor.Event.START_CONTAINER;
+import static com.amazon.ion.IonCursor.Event.START_SCALAR;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class IonCursorBinaryTest {
+
+    private static final IonBufferConfiguration STANDARD_BUFFER_CONFIGURATION = IonBufferConfiguration.Builder.standard().build();
+
+    private static Marker loadScalar(IonCursorBinary cursor) {
+        IonCursor.Event event = cursor.fillValue();
+        assertEquals(VALUE_READY, event);
+        Marker marker = cursor.getValueMarker();
+        assertNotNull(marker);
+        return marker;
+    }
+
+    private IonCursorBinary cursor = null;
+    private int numberOfIvmsEncountered = 0;
+
+    private final IvmNotificationConsumer countingIvmConsumer = (majorVersion, minorVersion) -> numberOfIvmsEncountered++;
+
+    @Before
+    public void setup() {
+        cursor = null;
+        numberOfIvmsEncountered = 0;
+    }
+
+    private void initializeCursor(int... data) {
+        cursor = new IonCursorBinary(STANDARD_BUFFER_CONFIGURATION, bytes(data), 0, data.length);
+        cursor.registerIvmNotificationConsumer(countingIvmConsumer);
+    }
+
+    @Test
+    public void basicContainer() {
+        initializeCursor(
+            0xE0, 0x01, 0x00, 0xEA,
+            0xD3, // Struct length 3
+            0x84, // Field SID 4
+            0x21, 0x01 // Int length 1, starting at byte index 7
+        );
+        assertEquals(START_CONTAINER, cursor.nextValue());
```

Comment:
Consider if some helper methods could give you a lightweight pattern for cutting the repetitiveness in this test:
```
   nextExpect(START_CONTAINER)
   stepInExpect(NEEDS_INSTRUCTION)
   expectMarker(7, 8)
```
  etc...

Reply:
Will do.